Source/WebCore/page/ContextMenuController.cpp

```diff
@@ -998,29 +1011,41 @@ void ContextMenuController::populate()
                 appendItem(StopItem, m_contextMenu.get());
                 appendItem(ReloadItem, m_contextMenu.get());
 #else
-                if (frame->page() && frame->page()->backForward().canGoBackOrForward(-1))
-                    appendItem(BackItem, m_contextMenu.get());
-
-                if (frame->page() && frame->page()->backForward().canGoBackOrForward(1))
-                    appendItem(ForwardItem, m_contextMenu.get());
 
-                // use isLoadingInAPISense rather than isLoading because Stop/Reload are
-                // intended to match WebKit's API, not WebCore's internal notion of loading status
-                if (loader.documentLoader()->isLoadingInAPISense())
-                    appendItem(StopItem, m_contextMenu.get());
-                else
-                    appendItem(ReloadItem, m_contextMenu.get());
+                
+                if (isMainFrame || isPDFDocument) {
+                    auto* framePage = frame->page();
+                    if (framePage && framePage->backForward().canGoBackOrForward(-1))
+                        appendItem(BackItem, m_contextMenu.get());
+
+                    if (framePage && framePage->backForward().canGoBackOrForward(1))
+                        appendItem(ForwardItem, m_contextMenu.get());
+                    
+                    // Here we use isLoadingInAPISense rather than isLoading because Stop/Reload are
+                    // intended to match WebKit's API, not WebCore's internal notion of loading status.
+                    if (loader.documentLoader()->isLoadingInAPISense())
+                        appendItem(StopItem, m_contextMenu.get());
+                    else
+                        appendItem(ReloadItem, m_contextMenu.get());
+                }
 #endif
                 }
 
-                if (frame->page() && !frame->isMainFrame())
+                if (ownerElement && !isMainFrame && !isPDFDocument) 
```

Comment:
Can you explain the reasoning behind this change? Should be restricted to `ENABLE(PDFJS)`?

Reply:
I talked to @nt1m, and he informed me that `isPDFDocument` is only true for PDFJS documents, so this is fine.